server/metadata/tenant.go

```diff
@@ -1051,6 +1051,33 @@ func (tenant *Tenant) ListProjects(_ context.Context) []string {
 	return projects
 }
 
+// DeleteTenant is used to delete tenant and all the content within it.
+// This needs to be followed up with a "restart" of server to clear memory state.
+// Be careful calling this.
+func (m *TenantManager) DeleteTenant(ctx context.Context, tx transaction.Tx, tenantToDelete *Tenant) error {
+	tenantToDelete.RLock()
+	defer tenantToDelete.RUnlock()
+	projects := tenantToDelete.ListProjects(ctx)
```

Comment:
ListProjects is taking a read lock on the tenant, so you don't need to grab a lock at the start of this method.

Reply:
good catch. fixed.